app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java

```diff
@@ -0,0 +1,171 @@
+            View view = getChildAt(i);
+            if (view instanceof AppBarLayout) {
+                appBarLayout = (AppBarLayout) view;
+                continue;
+            }
+            if (view instanceof NestedScrollView) {
+                scrollTop = !view.canScrollVertically(-1);
+                scrollEnd = !view.canScrollVertically(1);
+            }
+        }
+
+        if (appBarLayout == null || scrollTop && dyUnconsumed < 0 && isAppBarExpanded(appBarLayout)) {
+            translationY(-dyUnconsumed);
+        }
+        if (appBarLayout == null || scrollEnd && dyUnconsumed > 0 && isAppBarCollapsed(appBarLayout)) {
+            translationY(-dyUnconsumed);
+        }
+    }
+
+    @Override
+    public boolean onNestedPreFling(View target, float velocityX, float velocityY) {
+        return getY() != originalLocation.y || super.onNestedPreFling(target, velocityX, velocityY);
+    }
+
+    @Override
+    public void onStopNestedScroll(View target) {
+        super.onStopNestedScroll(target);
+        if (Math.abs(getY()) > overScrollThreshold && overScrollListener != null) {
+            float yTranslation;
+            if (getY() > 0) {
+                yTranslation = originalLocation.y + originalHeight;
+            } else {
+                yTranslation = originalLocation.y - originalHeight;
+            }
+            animate()
+                    .setDuration(getContext().getResources().getInteger(R.integer.activity_transition_mills))
+                    .setInterpolator(new AccelerateDecelerateInterpolator())
+                    .alpha(0)
+                    .translationY(yTranslation)
+                    .setListener(new Animator.AnimatorListener() {
+                        @Override
+                        public void onAnimationStart(Animator animation) {
+                        }
+
+                        @Override
+                        public void onAnimationEnd(Animator animation) {
+                            overScrollListener.onOverScroll();
+
+                        }
+
+                        @Override
+                        public void onAnimationCancel(Animator animation) {
+                        }
+
+                        @Override
+                        public void onAnimationRepeat(Animator animation) {
+                        }
+                    });
+            return;
+        }
+        if (getY() != originalLocation.y) {
+            animate()
+                    .setDuration(RESTORE_ANIM_DURATION)
+                    .setInterpolator(new AccelerateDecelerateInterpolator())
+                    .translationY(originalLocation.y)
+                    .scaleX(1)
+                    .scaleY(1);
+        }
+    }
+
+    private void translationY(float dy) {
+        setY(getY() + dy * 0.5f);
+    }
+
+    private boolean isAppBarExpanded(@NonNull AppBarLayout appBarLayout) {
+        return appBarLayout.getHeight() == appBarLayout.getBottom();
```

Comment:
> View#getHeight() is calculated by View#getBottom() - View#getTop()

I didn't know this, it may be able to replaced with View#getTop() == zero.
Thanks!